src/nibbles.rs

```diff
@@ -191,20 +201,111 @@ impl Nibbles {
         Self(SmallVec::with_capacity(capacity))
     }
 
-    /// Creates a new [`Nibbles`] instance from nibble bytes, without checking their validity.
+    /// Creates a new [`Nibbles`] instance by copying the given bytes.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the any of the bytes is not a valid nibble (`0..=0x0f`).
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use nybbles::Nibbles;
+    /// let nibbles = Nibbles::from_nibbles(&[0x0A, 0x0B, 0x0C, 0x0D]);
+    /// assert_eq!(nibbles[..], [0x0A, 0x0B, 0x0C, 0x0D]);
+    /// ```
+    ///
+    /// Invalid values will cause panics:
+    ///
+    /// ```should_panic
+    /// # use nybbles::Nibbles;
+    /// let nibbles = Nibbles::from_nibbles(&[0xFF]);
+    /// ```
+    #[inline]
+    #[track_caller]
+    pub fn from_nibbles<T: AsRef<[u8]>>(bytes: T) -> Self {
```

Comment:
For the checked versions I would encourage using an `Error` rather than panicking on invalid nibbles. 

e.g. for the following functions if any single bytes is `> 0x0f`
- `from_nibbles()`
- `from_vec()`

Although not strictly necessary this could be helpful for users decoding data from untrusted input, such that they don't need to validate the nibbles themselves. If they have validated the bytes they can use the `unchecked()` versions.